Deprecated EDAnalyzer in DTCalibration

As discussed here
https://github.com/cms-sw/cmssw/pull/35184#issuecomment-917138455
there are some deprecated EDAnalyzer in DTCalibration
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-422427/18493/llvm-analysis/legacy-mod-sa.txt

Will be follow-up by @swagata87 @namapane

```
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_1_X_2021-09-09-2300/src/CalibMuon/DTCalibration/plugins/DTTPAnalyzer.cc:22:1: warning: DTTPAnalyzer inherits from edm::EDProducer,edm::EDFilter,edm::EDAnalyzer, or edm::OutputModule [deprecated.edmChecker]
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_1_X_2021-09-09-2300/src/CalibMuon/DTCalibration/plugins/DTTTrigCalibration.h:33:1: warning: DTTTrigCalibration inherits from edm::EDProducer,edm::EDFilter,edm::EDAnalyzer, or edm::OutputModule [deprecated.edmChecker]
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_1_X_2021-09-09-2300/src/CalibMuon/DTCalibration/plugins/DTVDriftCalibration.h:35:1: warning: DTVDriftCalibration inherits from edm::EDProducer,edm::EDFilter,edm::EDAnalyzer, or edm::OutputModule [deprecated.edmChecker]
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_1_X_2021-09-09-2300/src/RecoLocalMuon/DTSegment/src/DTRecSegment2DExtendedProducer.h:34:1: warning: DTRecSegment2DExtendedProducer inherits from edm::EDProducer,edm::EDFilter,edm::EDAnalyzer, or edm::OutputModule [deprecated.edmChecker]
```

--- a/CalibMuon/DTCalibration/plugins/DTTPAnalyzer.cc
+++ b/CalibMuon/DTCalibration/plugins/DTTPAnalyzer.cc
@@ -3,7 +3,7 @@
  *  \author A. Vilela Pereira
  */
 
-#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
 #include "FWCore/Framework/interface/ConsumesCollector.h"
 #include "FWCore/Utilities/interface/InputTag.h"
 #include "FWCore/Framework/interface/ESHandle.h"
@@ -19,7 +19,7 @@ class DTGeometry;
 class DTTTrigBaseSync;
 class TFile;
 
-class DTTPAnalyzer : public edm::EDAnalyzer {
+class DTTPAnalyzer : public edm::one::EDAnalyzer<> {
 public:
   DTTPAnalyzer(const edm::ParameterSet&);
   ~DTTPAnalyzer() override;

--- a/CalibMuon/DTCalibration/plugins/DTTTrigCalibration.h
+++ b/CalibMuon/DTCalibration/plugins/DTTTrigCalibration.h
@@ -9,7 +9,7 @@
  *  \author G. Cerminara - INFN Torino
  */
 
-#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
 #include "DataFormats/MuonDetId/interface/DTSuperLayerId.h"
 #include "DataFormats/MuonDetId/interface/DTLayerId.h"
 
@@ -30,7 +30,7 @@ class DTTtrig;
 class DTStatusFlag;
 class DTStatusFlagRcd;
 
-class DTTTrigCalibration : public edm::EDAnalyzer {
+class DTTTrigCalibration : public edm::one::EDAnalyzer<> {
 public:
   /// Constructor
   DTTTrigCalibration(const edm::ParameterSet& pset);

--- a/CalibMuon/DTCalibration/plugins/DTVDriftCalibration.h
+++ b/CalibMuon/DTCalibration/plugins/DTVDriftCalibration.h
@@ -7,7 +7,7 @@
  *  \author M. Giunta
  */
 
-#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
 #include "FWCore/Utilities/interface/InputTag.h"
 #include "DataFormats/MuonDetId/interface/DTWireId.h"
 #include "DataFormats/DTRecHit/interface/DTRecSegment4DCollection.h"
@@ -32,7 +32,7 @@ class DTMeanTimerFitter;
 class DTGeometry;
 class MuonGeometryRecord;
 
-class DTVDriftCalibration : public edm::EDAnalyzer {
+class DTVDriftCalibration : public edm::one::EDAnalyzer<> {
 public:
   /// Constructor
   DTVDriftCalibration(const edm::ParameterSet& pset);

--- a/CalibMuon/DTCalibration/test/DBTools/DumpDBToFile.cc
+++ b/CalibMuon/DTCalibration/test/DBTools/DumpDBToFile.cc
@@ -15,22 +15,6 @@
 #include "FWCore/Utilities/interface/Exception.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
-#include "CondFormats/DTObjects/interface/DTMtime.h"
-#include "CondFormats/DataRecord/interface/DTMtimeRcd.h"
-#include "CondFormats/DTObjects/interface/DTTtrig.h"
-#include "CondFormats/DataRecord/interface/DTTtrigRcd.h"
-#include "CondFormats/DTObjects/interface/DTT0.h"
-#include "CondFormats/DataRecord/interface/DTT0Rcd.h"
-#include "CondFormats/DataRecord/interface/DTStatusFlagRcd.h"
-#include "CondFormats/DTObjects/interface/DTStatusFlag.h"
-#include "CondFormats/DataRecord/interface/DTDeadFlagRcd.h"
-#include "CondFormats/DTObjects/interface/DTDeadFlag.h"
-#include "CondFormats/DataRecord/interface/DTReadOutMappingRcd.h"
-#include "CondFormats/DTObjects/interface/DTReadOutMapping.h"
-#include "CondFormats/DTObjects/interface/DTRecoConditions.h"
-#include "CondFormats/DataRecord/interface/DTRecoConditionsTtrigRcd.h"
-#include "CondFormats/DataRecord/interface/DTRecoConditionsVdriftRcd.h"
-#include "CondFormats/DataRecord/interface/DTRecoConditionsUncertRcd.h"
 #include <iostream>
 #include <fstream>
 #include <algorithm>
@@ -53,6 +37,30 @@ DumpDBToFile::DumpDBToFile(const ParameterSet& pset) {
 
   if (format != "Legacy" && format != "DTRecoConditions")
     throw cms::Exception("IncorrectSetup") << "Parameter format is not valid, check the cfg file" << endl;
+
+  if (dbToDump == "VDriftDB") {
+    if (format == "Legacy") {
+      mTimeMapToken_ = esConsumes<edm::Transition::BeginRun>();
+    } else {
+      vDriftToken_ = esConsumes<edm::Transition::BeginRun>();
+    }
+  } else if (dbToDump == "TTrigDB") {
+    if (format == "Legacy") {
+      tTrigMapToken_ = esConsumes<edm::Transition::BeginRun>();
+    } else {
+      tTrigToken_ = esConsumes<edm::Transition::BeginRun>();
+    }
+  } else if (dbToDump == "TZeroDB") {
+    t0MapToken_ = esConsumes<edm::Transition::BeginRun>();
+  } else if (dbToDump == "NoiseDB") {
+    statusMapToken_ = esConsumes<edm::Transition::BeginRun>();
+  } else if (dbToDump == "DeadDB") {
+    deadMapToken_ = esConsumes<edm::Transition::BeginRun>();
+  } else if (dbToDump == "ChannelsDB") {
+    readOutMapToken_ = esConsumes<edm::Transition::BeginRun>();
+  } else if (dbToDump == "RecoUncertDB") {
+    uncertToken_ = esConsumes<edm::Transition::BeginRun>();
+  }
 }
 
 DumpDBToFile::~DumpDBToFile() {}
@@ -61,44 +69,26 @@ void DumpDBToFile::beginRun(const edm::Run&, const EventSetup& setup) {
   // Read the right DB accordingly to the parameter dbToDump
   if (dbToDump == "VDriftDB") {
     if (format == "Legacy") {
-      ESHandle<DTMtime> mTime;
-      setup.get<DTMtimeRcd>().get(mTime);
-      mTimeMap = &*mTime;
+      mTimeMap = &setup.getData(mTimeMapToken_);
     } else if (format == "DTRecoConditions") {
-      ESHandle<DTRecoConditions> h_rconds;
-      setup.get<DTRecoConditionsVdriftRcd>().get(h_rconds);
-      rconds = &*h_rconds;
+      rconds = &setup.getData(vDriftToken_);
     }
   } else if (dbToDump == "TTrigDB") {
     if (format == "Legacy") {
-      ESHandle<DTTtrig> tTrig;
-      setup.get<DTTtrigRcd>().get(dbLabel, tTrig);
-      tTrigMap = &*tTrig;
+      tTrigMap = &setup.getData(tTrigMapToken_);
     } else if (format == "DTRecoConditions") {
-      ESHandle<DTRecoConditions> h_rconds;
-      setup.get<DTRecoConditionsTtrigRcd>().get(h_rconds);
-      rconds = &*h_rconds;
+      rconds = &setup.getData(tTrigToken_);
     }
   } else if (dbToDump == "TZeroDB") {
-    ESHandle<DTT0> t0;
-    setup.get<DTT0Rcd>().get(t0);
-    tZeroMap = &*t0;
+    tZeroMap = &setup.getData(t0MapToken_);
   } else if (dbToDump == "NoiseDB") {
-    ESHandle<DTStatusFlag> status;
-    setup.get<DTStatusFlagRcd>().get(status);
-    statusMap = &*status;
+    statusMap = &setup.getData(statusMapToken_);
   } else if (dbToDump == "DeadDB") {
-    ESHandle<DTDeadFlag> dead;
-    setup.get<DTDeadFlagRcd>().get(dead);
-    deadMap = &*dead;
+    deadMap = &setup.getData(deadMapToken_);
   } else if (dbToDump == "ChannelsDB") {
-    ESHandle<DTReadOutMapping> channels;
-    setup.get<DTReadOutMappingRcd>().get(channels);
-    channelsMap = &*channels;
+    channelsMap = &setup.getData(readOutMapToken_);
   } else if (dbToDump == "RecoUncertDB") {
-    ESHandle<DTRecoConditions> h_rconds;
-    setup.get<DTRecoConditionsUncertRcd>().get(h_rconds);
-    rconds = &*h_rconds;
+    rconds = &setup.getData(uncertToken_);
   }
 }
 
@@ -237,7 +227,7 @@ void DumpDBToFile::endJob() {
         float t0rms;
         // t0s and rms are TDC counts
         tZeroMap->get(wireId, t0mean, t0rms, DTTimeUnits::counts);
-        cout << wireId << " TZero mean (TDC counts): " << t0mean << " TZero RMS (TDC counts): " << t0rms << endl;
+        //        cout << wireId << " TZero mean (TDC counts): " << t0mean << " TZero RMS (TDC counts): " << t0rms << endl;
         vector<float> consts;
         consts.push_back(-1);
         consts.push_back(-1);

--- a/CalibMuon/DTCalibration/test/DBTools/DumpDBToFile.h
+++ b/CalibMuon/DTCalibration/test/DBTools/DumpDBToFile.h
@@ -12,16 +12,26 @@
 #include "FWCore/Framework/interface/Frameworkfwd.h"
 #include "FWCore/Framework/interface/one/EDAnalyzer.h"
 
+#include "CondFormats/DTObjects/interface/DTMtime.h"
+#include "CondFormats/DataRecord/interface/DTMtimeRcd.h"
+#include "CondFormats/DTObjects/interface/DTTtrig.h"
+#include "CondFormats/DataRecord/interface/DTTtrigRcd.h"
+#include "CondFormats/DTObjects/interface/DTT0.h"
+#include "CondFormats/DataRecord/interface/DTT0Rcd.h"
+#include "CondFormats/DataRecord/interface/DTStatusFlagRcd.h"
+#include "CondFormats/DTObjects/interface/DTStatusFlag.h"
+#include "CondFormats/DataRecord/interface/DTDeadFlagRcd.h"
+#include "CondFormats/DTObjects/interface/DTDeadFlag.h"
+#include "CondFormats/DataRecord/interface/DTReadOutMappingRcd.h"
+#include "CondFormats/DTObjects/interface/DTReadOutMapping.h"
+#include "CondFormats/DTObjects/interface/DTRecoConditions.h"
+#include "CondFormats/DataRecord/interface/DTRecoConditionsTtrigRcd.h"
+#include "CondFormats/DataRecord/interface/DTRecoConditionsVdriftRcd.h"
+#include "CondFormats/DataRecord/interface/DTRecoConditionsUncertRcd.h"
+
 #include <string>
 
-class DTMtime;
-class DTTtrig;
-class DTT0;
-class DTStatusFlag;
-class DTDeadFlag;
 class DTCalibrationMap;
-class DTReadOutMapping;
-class DTRecoConditions;
 
 class DumpDBToFile : public edm::one::EDAnalyzer<edm::one::WatchRuns> {
 public:
@@ -42,6 +52,16 @@ public:
 
 protected:
 private:
+  edm::ESGetToken<DTMtime, DTMtimeRcd> mTimeMapToken_;
+  edm::ESGetToken<DTTtrig, DTTtrigRcd> tTrigMapToken_;
+  edm::ESGetToken<DTT0, DTT0Rcd> t0MapToken_;
+  edm::ESGetToken<DTStatusFlag, DTStatusFlagRcd> statusMapToken_;
+  edm::ESGetToken<DTDeadFlag, DTDeadFlagRcd> deadMapToken_;
+  edm::ESGetToken<DTReadOutMapping, DTReadOutMappingRcd> readOutMapToken_;
+  edm::ESGetToken<DTRecoConditions, DTRecoConditionsTtrigRcd> tTrigToken_;
+  edm::ESGetToken<DTRecoConditions, DTRecoConditionsVdriftRcd> vDriftToken_;
+  edm::ESGetToken<DTRecoConditions, DTRecoConditionsUncertRcd> uncertToken_;
+
   const DTMtime *mTimeMap;
   const DTTtrig *tTrigMap;
   const DTT0 *tZeroMap;

--- a/CalibMuon/DTCalibration/test/DBTools/DumpDBToFile.h
+++ b/CalibMuon/DTCalibration/test/DBTools/DumpDBToFile.h
@@ -10,7 +10,7 @@
  */
 
 #include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
 
 #include <string>
 
@@ -23,20 +23,22 @@ class DTCalibrationMap;
 class DTReadOutMapping;
 class DTRecoConditions;
 
-class DumpDBToFile : public edm::EDAnalyzer {
+class DumpDBToFile : public edm::one::EDAnalyzer<edm::one::WatchRuns> {
 public:
   /// Constructor
   DumpDBToFile(const edm::ParameterSet &pset);
 
   /// Destructor
-  virtual ~DumpDBToFile();
+  ~DumpDBToFile() override;
 
   // Operations
-  virtual void beginRun(const edm::Run &run, const edm::EventSetup &setup);
+  void beginRun(const edm::Run &run, const edm::EventSetup &setup) override;
 
-  virtual void analyze(const edm::Event &event, const edm::EventSetup &setup) {}
+  void endRun(const edm::Run &run, const edm::EventSetup &setup) override {}
 
-  virtual void endJob();
+  void analyze(const edm::Event &event, const edm::EventSetup &setup) override {}
+
+  void endJob() override;
 
 protected:
 private:

--- a/CalibMuon/DTCalibration/test/DBTools/DumpFileToDB.h
+++ b/CalibMuon/DTCalibration/test/DBTools/DumpFileToDB.h
@@ -10,7 +10,7 @@
  */
 
 #include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
 
 #include <string>
 #include <fstream>
@@ -19,21 +19,22 @@
 class DTCalibrationMap;
 class DTTtrig;
 
-class DumpFileToDB : public edm::EDAnalyzer {
+class DumpFileToDB : public edm::one::EDAnalyzer<edm::one::WatchRuns> {
 public:
   /// Constructor
   DumpFileToDB(const edm::ParameterSet& pset);
 
   /// Destructor
-  virtual ~DumpFileToDB();
+  ~DumpFileToDB() override;
 
   // Operations
-  // Operations
-  virtual void beginRun(const edm::Run& run, const edm::EventSetup& setup);
+  void beginRun(const edm::Run& run, const edm::EventSetup& setup) override;
 
-  virtual void analyze(const edm::Event& event, const edm::EventSetup& setup) {}
+  void endRun(const edm::Run& run, const edm::EventSetup& setup) override {}
 
-  virtual void endJob();
+  void analyze(const edm::Event& event, const edm::EventSetup& setup) override {}
+
+  void endJob() override;
 
 protected:
 private:

--- a/CalibMuon/DTCalibration/test/DumpDBToFile_cfg.py
+++ b/CalibMuon/DTCalibration/test/DumpDBToFile_cfg.py
@@ -14,7 +14,7 @@ options.register('type',
                  'TTrigDB', #default value
                  VarParsing.VarParsing.multiplicity.singleton,
                  VarParsing.VarParsing.varType.string,
-                 "Database to read: 'TZeroDB', 'TTrigDB',  'VDriftDB',  or 'UncertDB'")
+                 "Database to read: 'TZeroDB', 'TTrigDB',  'VDriftDB', 'UncertDB', 'NoiseDB', 'DeadDB', 'ChannelsDB'")
 
 options.register('GT',
                  'auto:run2_data', #default value
@@ -58,7 +58,7 @@ if DBFORMAT not in ['Legacy', 'DTRecoConditions'] :
     print('\nERROR: invalid value for dbformat: ',  DBFORMAT,'\n')
     exit()
     
-if TYPE not in ['TZeroDB', 'TTrigDB',  'VDriftDB', 'UncertDB'] :
+if TYPE not in ['TZeroDB', 'TTrigDB',  'VDriftDB', 'UncertDB', 'NoiseDB', 'DeadDB', 'ChannelsDB'] :
     print('\nERROR: invalid value for type: ',  TYPE,'\n')
     exit()
 
@@ -73,7 +73,10 @@ if INPUTTAG!="" and INPUTFILE!="" :
 ofExt = {'TZeroDB'  : '_t0.txt',
          'TTrigDB'  : '_ttrig.txt',
          'VDriftDB' : '_vdrift.txt',
-         'UncertDB' : '_uncert.txt'}
+         'UncertDB' : '_uncert.txt',
+         'NoiseDB'  : '_noise.txt',
+         'DeadDB'   : '_dead.txt',
+         'ChannelsDB' : '_channels.txt'}
 
 
 if INPUTFILE!="":
@@ -101,6 +104,10 @@ elif DBFORMAT=="DTRecoConditions" :
     if TYPE=="VDriftDB" : RECORD = "DTRecoConditionsVdriftRcd"
     if TYPE=="UncertDB" : RECORD = "DTRecoConditionsUncertRcd"
 
+
+if TYPE == 'NoiseDB' : RECORD = 'DTStatusFlagRcd'
+elif TYPE == 'DeadDB' : RECORD = 'DTDeadFlagRcd'
+elif TYPE == 'ChannelsDB' : RECORD = 'DTReadOutMappingRcd'
 
 process = cms.Process("DumpDBToFile")
 process.load("CondCore.CondDB.CondDB_cfi")
@@ -203,10 +210,47 @@ process.dumpUncertToFile = cms.EDAnalyzer("DumpDBToFile",
     outputFileName = cms.untracked.string(OUTPUTFILE)
 )
 
+process.dumpNoiseToFile = cms.EDAnalyzer("DumpDBToFile",
+    dbToDump = cms.untracked.string('NoiseDB'),
+    dbLabel = cms.untracked.string(''),
+    dbFormat = cms.untracked.string(DBFORMAT),
+    calibFileConfig = cms.untracked.PSet(
+        nFields = cms.untracked.int32(8),
+        calibConstGranularity = cms.untracked.string('byWire')
+    ),
+    outputFileName = cms.untracked.string(OUTPUTFILE)
+)
 
-if TYPE=="TZeroDB" :     process.p2 = cms.Path(process.dumpT0ToFile)
-if TYPE=="TTrigDB" :     process.p2 = cms.Path(process.dumpTTrigToFile)
-if TYPE=="VDriftDB" :    process.p2 = cms.Path(process.dumpVdToFile)
-if TYPE=="UncertDB": process.p2 = cms.Path(process.dumpUncertToFile)
+process.dumpDeadToFile = cms.EDAnalyzer("DumpDBToFile",
+    dbToDump = cms.untracked.string('DeadDB'),
+    dbLabel = cms.untracked.string(''),
+    dbFormat = cms.untracked.string(DBFORMAT),
+    calibFileConfig = cms.untracked.PSet(
+        nFields = cms.untracked.int32(8),
+        calibConstGranularity = cms.untracked.string('byWire')
+    ),
+    outputFileName = cms.untracked.string(OUTPUTFILE)
+)
+
+process.dumpChannelsToFile = cms.EDAnalyzer("DumpDBToFile",
+    dbToDump = cms.untracked.string('ChannelsDB'),
+    dbLabel = cms.untracked.string(''),
+    dbFormat = cms.untracked.string(DBFORMAT),
+    calibFileConfig = cms.untracked.PSet(
+        nFields = cms.untracked.int32(8),
+        calibConstGranularity = cms.untracked.string('byWire')
+    ),
+    outputFileName = cms.untracked.string(OUTPUTFILE)
+)
+
+
+
+if   TYPE=="TZeroDB" :     process.p2 = cms.Path(process.dumpT0ToFile)
+elif TYPE=="TTrigDB" :     process.p2 = cms.Path(process.dumpTTrigToFile)
+elif TYPE=="VDriftDB" :    process.p2 = cms.Path(process.dumpVdToFile)
+elif TYPE=="UncertDB":     process.p2 = cms.Path(process.dumpUncertToFile)
+elif TYPE=='NoiseDB' :     process.p2 = cms.Path(process.dumpNoiseToFile)
+elif TYPE=='DeadDB' :      process.p2 = cms.Path(process.dumpDeadToFile)
+elif TYPE=='ChannelsDB' :  process.p2 = cms.Path(process.dumpChannelsToFile)
 
 

--- a/RecoLocalMuon/DTSegment/src/DTRecSegment2DExtendedProducer.h
+++ b/RecoLocalMuon/DTSegment/src/DTRecSegment2DExtendedProducer.h
@@ -11,7 +11,7 @@
  */
 
 /* Base Class Headers */
-#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
 #include "DataFormats/DTRecHit/interface/DTRecHitCollection.h"
 #include "DataFormats/DTRecHit/interface/DTRecClusterCollection.h"
 
@@ -31,7 +31,7 @@ class MuonGeometryRecord;
 
 /* Class DTRecSegment2DExtendedProducer Interface */
 
-class DTRecSegment2DExtendedProducer : public edm::EDProducer {
+class DTRecSegment2DExtendedProducer : public edm::stream::EDProducer<> {
 public:
   /// Constructor
   DTRecSegment2DExtendedProducer(const edm::ParameterSet&);

